src/decorators.py

```diff
@@ -51,16 +51,15 @@ def wrap1(func):
         @wraps(func)
         def wrap2(_pname=None, *args, **kwargs):
             pname = os.environ.get('PROJECT', _pname)
-            if not pname or not pname.strip():
-                project_list = project.filtered_projects(filterfn)
-                #project_list = project.project_list()
+            project_list = project.filtered_projects(filterfn)
+            if not pname or not pname.strip() or pname not in project_list:
                 pname = utils._pick("project", sorted(project_list), default_file=deffile('.project'))
             return func(pname, *args, **kwargs)
         return wrap2
     return wrap1
 
 #pylint: disable=invalid-name
-requires_branch_deployable_project = requires_filtered_project(lambda pname, project: project.has_key('repo'))
+requires_branch_deployable_project = requires_filtered_project(lambda pname, project: project.has_key('repo') and project.get('repo'))
```

Comment:
so - in this case, the project is specifying a `repo`, but the `repo` value evaluates to None? changing this to simply project.get_key